server/src/utils/callAutomationUtils.ts

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  CallAutomationClient,
+  CallLocator,
+  StreamingData,
+  TranscriptionData,
+  TranscriptionMetadata,
+  TranscriptionOptions
+} from '@azure/communication-call-automation';
+import { ConversationSummaryInput } from './summarizationUtils';
+import { CommunicationUserIdentifier } from '@azure/communication-common';
+import { getServerConfig } from './getConfig';
+
+export interface CallTranscription {
+  metadata: TranscriptionMetadata;
+  data: TranscriptionData[];
+}
+
+const callAutomationConfig = getServerConfig().callAutomation;
+
+/**
+ * Call automation config is required for call automation to work.
+ * if this is undefined then each of the values that are undefined will be
+ * replaced with empty strings.
+ *
+ * Make sure that you either have your call automation config set in the environment
+ * variables or in the config file.
+ */
+if (!callAutomationConfig) {
+  console.warn('Call automation config is not set');
+}
+
+let callAutomationClient: CallAutomationClient | undefined = undefined;
+export const getCallAutomationClient = (): CallAutomationClient =>
+  callAutomationClient ??
+  (callAutomationClient = new CallAutomationClient(getServerConfig().communicationServicesConnectionString));
+
+export const connectRoomsCallWithTranscription = async (roomId: string): Promise<void> => {
+  const transcriptionOptions = {
+    transportUrl: callAutomationConfig?.ServerWebSocketUrl ?? '',
+    transportType: 'websocket',
+    locale: 'en-US',
+    startTranscription: true
+  };
+
+  const options = {
+    callIntelligenceOptions: {
+      cognitiveServicesEndpoint: callAutomationConfig?.CognitionAPIEndpoint
+    },
+    transcriptionOptions: transcriptionOptions
+  };
+
+  const callbackUri = callAutomationConfig?.ServerHttpUrl;
+
+  /**
+   * Call automation needs to create the call.
+   */
+
+  const automationClient = getCallAutomationClient();
+  const roomsLocator: CallLocator = { kind: 'roomCallLocator', id: roomId };
+  const connectedCallResult = await automationClient.connectCall(roomsLocator, callbackUri ?? '', options);
+  console.log((await connectedCallResult.callConnection.getCallConnectionProperties()).serverCallId);
```

Comment:
Was this console just for debugging?

Reply:
yes a lot of them are will make sure to clean up anything not needed but some are useful for developers to see when running the sample